### ISSUE TITLE
Keys with dynamic scope arguments in views are parsed as if there were no scope argument

### ISSUE THREAD

**Issue by shaneog** [glebm/i18n-tasks#213]
Firstly, sorry for opening a support issue here!

Our app uses translations using the explicitly scoped, symbols form, e.g.
```ruby
I18n.t(:button_text, scope: [:search, :header_form])
```
We have a lot of this code in our large codebase.

We also use some dynamic keys, e.g.
```ruby
I18n.t(:title_no_results, scope: [:search, params[:action]], query: params[:q])
```

I've tried using the following hint using `i18n-tasks-use` but it doesn't work using either of the below comments on the preceding line
```haml
/ i18n-tasks-use t('search.%{params[:action]}. title_no_results')
/ i18n-tasks-use t('search.%{params[:action]}. title_no_results', query: params[:q])
```

So right now I can't use the `missing` command or RSpec test because of false positives.
Am I doing something wrong or is this type of dynamic key not supported?

**Comment by glebm** [comment]
Dynamic keys need to be listed explicitly, so e.g. if the possible actions are `foo` and `bar`, the above should be:

```slim
/ i18n-tasks-use t('search.foo.title')
/ i18n-tasks-use t('search.bar.title')
```

**Comment by shaneog** [comment]
Thank you for the quick response.

> In the view files, there is an additional restriction that scope must be the last argument.

Since this is used in a view, does that mean that it is not supported?
```haml
= t(:title_no_results, scope: [:search, params[:action]], query: params[:q])
```
I'll will use the explicit keys in the `i18n-tasks-use` comment, and report back.

**Comment by shaneog** [comment]
Ok, I've tried using the below and it didn't work.

```haml
  / i18n-tasks-use t('search.foo.title_no_results')
  / i18n-tasks-use t('search.bar.title_no_results')
  / i18n-tasks-use t('search.foobar.title_no_results')
  / i18n-tasks-use t('search.barfoo.title_no_results')
  %h2.no-results-title= t(:title_no_results, query: params[:q], scope: [:search, params[:action]])
```

I get the following output when I run `i18n-tasks missing`:
```sh
Missing translations (1) | i18n-tasks v0.9.6
+--------+-------------------------+-------------------------------------------+
| Locale | Key                     | Value in other locales or source          |
+--------+-------------------------+-------------------------------------------+
|  all   | title_no_results_html   | app/views/search/_no_results.html.haml:10 |
+--------+-------------------------+-------------------------------------------+

```

Any ideas?

**Comment by glebm** [comment]
The output of `i18n-tasks` missing looks like it is for an unrelated key, title_no_results*_html*

**Comment by glebm** [comment]
Sorry, when I said the *last* argument, I actually meant to say the *first* argument after the key!

**Comment by shaneog** [comment]
> The output of i18n-tasks missing looks like it is for an unrelated key, title_no_results*_html*

Sorry, that was me redacting some key names. In that example it should have been `title_no_results`.

There are hundreds of these examples in my codebase, but I've excluded all other files except this one for now to isolate what is happening. So it's definitely this key 😕 
---

> Sorry, when I said the last argument, I actually meant to say the first argument after the key!

So it should be this?
```haml
  %h2.no-results-title= t(:title_no_results, scope: [:search, params[:action]]), query: params[:q]
```

I've just tried it that way too and it's still not working I'm afraid.

**Comment by glebm** [comment]
I've just look at the [code for this](https://github.com/glebm/i18n-tasks/blob/master/lib/i18n/tasks/scanners/pattern_with_scope_scanner.rb), and looks like the scope argument is completely ignored if it doesn't consist of literals only.

So you could submit a fix that correctly parses such scope arguments as e.g. `"title_no_results.search.#{params[:action]}"` or alternatively completely ignores the entire key if they have a dynamic scope argument.

Alternatively, in your codebase, you could refactor such keys to not use a scope argument.

**Comment by glebm** [comment]
Fix released in v0.9.9.
